pkg/agent/netpol/utils.go

```diff
@@ -537,3 +539,19 @@ func GetNodeIP(node *apiv1.Node) (net.IP, error) {
 	}
 	return nil, errors.New("host IP unknown")
 }
+
+// CacheSync performs cache synchronization under timeout limit
+func CacheSyncOrTimeout(informerFactory informers.SharedInformerFactory, stopCh <-chan struct{}, cacheSyncTimeout time.Duration) error {
+	syncOverCh := make(chan struct{})
+	go func() {
+		informerFactory.WaitForCacheSync(stopCh)
+		close(syncOverCh)
+	}()
+
+	select {
+	case <-time.After(cacheSyncTimeout):
+		return errors.New(cacheSyncTimeout.String() + " timeout")
```

Comment:
Can we indicate what it was that timed out, rather than just how long the timeout was?

Reply:
This is just a copy of kube-proxy code that was changed a little to take a duration as parameter instead of from config. 
The usage is pre-pended here: https://github.com/rancher/k3s/pull/2363/files#diff-d9107b0f6a0c5d46b9e79b55e7de5df5R1687